content/Secure Coding/Buffer Overflow/CS0 C++/checklist.js

```diff
@@ -16,115 +16,147 @@ $(document).ready(function() {
-
-					waitingOn = [];
-					$(".span-"+name).each(function(index) {
-						if( $(this).hasClass(next+"-"+name) ) {
-							waitingOn.push(index);
-						}
-					});
+			});
+
+			showPopOver(spans[7]); // show Popover over the index span
+			var tag = "Pgm1CL-var-range"; //the one that needs the popover
+			
+			$('input[name=popSave2]').on('click', function() {
+				if ($('#lowInt').val() == "0" && $('#popSelect1').val() == "<="
+					&& $('#popSelect2').val() == "<" && $('#highInt').val() == "10"
+				) { // these need to change to the right answer for the question
+					$('.rangepop').hide();
+					$(".span-"+name).click(); //click the empty span so that next question is asked.
+					$('.popover').css('background-color', 'white');
+				} else {
+					$('.popover').css('background-color', 'lightpink');
 				}
+			});
+		} else { //NORMAL QUESTION
 
-					return next;
+			waitingOn = [];
+			$(".span-"+name).each(function(index) {
+				if( $(this).hasClass(next+"-"+name) ) {
+					waitingOn.push(index);
 				}
+			});
+		}	
+		//removes highlighted sections so user is not seeing correct answers of previous question
+		$(".span-" + name).removeClass("si-code-clicked");
+		return next;
+	}
 
+	current = advance();
 
-				current = advance();
-
-
+	//Handles score for each question
 	$(".span-"+name).each(function(index) {
 		spans.push($(this));
-		//console.log(spans);
+		
 		$(this).on('click', function() {
 			span = $(this);
 
 			// continue if user is currently supposed to click this span, and hasn't already
-			if( span.hasClass(current+"-"+name) && $.inArray(index,clicked) < 0 ) {
-				clicked.push(index);					// 		note that it's been clicked, programmatically
-				span.addClass("si-code-clicked");		//		note that it's been clicked, graphically
-				if(span.hasClass(name+"-var-modify-"+name)|| span.hasClass(name+"-var-array-"+name)) { //needs to be changed for vulnerabilities
-					span.addClass("si-code-vulnerability"); //	some spans get extra graphics to indicate vulnerability
-				}
-
+			if(span.hasClass(current+"-"+name) && $.inArray(index,clicked) < 0 ) {
+				clicked.push(index);  // note that it's been clicked, programmatically
+				span.addClass("si-code-clicked");  // note that it's been clicked, graphically
 
 				// Check if 'current' question is finished yet
 				var finished = true;
+
+				// logic for increasing the progress bar
+				let currentProgress = ((clicked.length - previousWaitingOnCount)/waitingOn.length)*100;
+				document.getElementById(current + "-progress-data-label").style.width = `${currentProgress}%`;
+
 				for(i in waitingOn) {
 					finished &= $.inArray(waitingOn[i],clicked) >= 0;
 				}
 
 				// if it is, go to next question
-				if(finished) current = advance(current);
+				if (finished) {
+					// highlights correct answer before moving on to next question
+					setTimeout(function(){
+						previousWaitingOnCount += waitingOn.length;
+						current = advance(current);
+					}, 500);
+				}
```

Comment:
If you have a question that only needs you to click one span, the span does not get highlighted before the next question becomes active. You'll need to add this timeout so the user can quickly see that they clicked the right answer.

This goes hand-in-hand with 
`//removes highlighted sections so user is not seeing correct answers of previous question
$(".span-" + name).removeClass("si-code-clicked");`
If the checklist has only multi-span questions, this code can go at the end of `advance()` above the return statement.
If the checklist contains single-span questions, this code goes at the beginning of `advance()` under `var next;`